From 8570aaa9687f179967ad493e4f1dd8ace43869c9 Mon Sep 17 00:00:00 2001
From: Jay Cornwall <Jay.Cornwall@amd.com>
Date: Wed, 22 Feb 2017 11:04:50 -0600
Subject: [PATCH 5330/5855] drm/amdkfd: Fix Gfx9 AQL MQD initialization

CP_HQD_AQL_CONTROL.CONTROL0_EN should not be set in the initial
MQD state. Doing so can cause intermittent failures in barrier
packet processing.

Clear CP_HQD_AQL_CONTROL.CONTROL0_EN in the initial MQD state.

Signed-off-by: Jay Cornwall <Jay.Cornwall@amd.com>
Reviewed-by: Felix Kuehling <Felix.Kuehling@amd.com>
---
 drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_v9.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_v9.c b/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_v9.c
index 3caeb6e..c80691e 100644
--- a/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_v9.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_v9.c
@@ -148,8 +148,7 @@ static int init_mqd(struct mqd_manager *mm, void **mqd,
 
 	if (q->format == KFD_QUEUE_FORMAT_AQL) {
 		m->cp_hqd_aql_control =
-			1 << CP_HQD_AQL_CONTROL__CONTROL0__SHIFT |
-			1 << CP_HQD_AQL_CONTROL__CONTROL0_EN__SHIFT;
+			1 << CP_HQD_AQL_CONTROL__CONTROL0__SHIFT;
 	}
 
 	if (q->tba_addr) {
-- 
2.7.4

